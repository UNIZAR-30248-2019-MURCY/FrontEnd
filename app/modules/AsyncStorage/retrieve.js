import { AsyncStorage } from 'react-native';

export const retrieveItem = async (key) => {
    let data ='';
    try {
        data = await AsyncStorage.getItem(key)
        console.log(data)
    } catch (error) {
        console.log(error.message);
    }
    return data;
}
