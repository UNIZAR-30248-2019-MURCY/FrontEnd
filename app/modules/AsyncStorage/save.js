import { AsyncStorage } from 'react-native';

export const saveData = async (key, data) => {
    try {
        await AsyncStorage.setItem(key, JSON.stringify(data));
        console.log('data stored');
    } catch (error) {
        // Error retrieving data
        console.log(error.message);
    }
};
