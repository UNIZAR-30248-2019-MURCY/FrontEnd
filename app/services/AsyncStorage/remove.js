import { AsyncStorage } from 'react-native';

export const removeItem = async (key) => {
    try {
        await AsyncStorage.removeItem(key);
        console.log('data removed');
        return true;
    } catch (error) {
        console.log(error.message);
        return false;
    }
}
