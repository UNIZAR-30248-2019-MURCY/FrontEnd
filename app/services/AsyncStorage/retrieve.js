import { AsyncStorage } from 'react-native';

export const retrieveItem = async (key) => {
    let data ='';
    try {
        let dataStored = await AsyncStorage.getItem(key)
        data = JSON.parse(dataStored).token;
        console.log('retrievedItem')
    } catch (error) {
        console.log(error.message);
    }
    return data;
}
