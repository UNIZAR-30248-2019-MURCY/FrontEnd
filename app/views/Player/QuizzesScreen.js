import React, {Component} from 'react';
import {
    View,
    StyleSheet, Platform, ActivityIndicator, Animated, TextInput, TouchableOpacity
} from 'react-native'
import {Button, CheckBox, colors, Text} from 'react-native-elements';
import Carousel from 'react-native-snap-carousel';
import {sliderWidth, itemWidth} from '../../modules/QuizzesCards/SliderEntry.style';
import SliderEntry from '../../components/SliderEntry';
import {ENTRIES1, ENTRIES2} from '../../modules/QuizzesCards/entries';
import Icon from 'react-native-vector-icons/FontAwesome';
import { scrollInterpolators, animatedStyles } from '../../modules/QuizzesCards/animations';
import {retrieveItem} from "../../modules/AsyncStorage/retrieve";
import {listQuestions} from "../../services/question/questionFuncs";
import {quizSearch} from "../../services/quiz/quizFuncs";
import ActionButton from "react-native-action-button";
import {removeAll} from "../../modules/AsyncStorage/remove";


const WEB = Platform.OS === 'web';

export default class QuizzesScreen extends Component {

    constructor(props) {
        super(props);
        this.state = {
            lastColor: '',
            loading: false,
            data: [],
            showFilter: false,
            ready: false,
            SlideInLeft: new Animated.Value(0),
            slideUpValue: new Animated.Value(0),
            fadeValue: new Animated.Value(0),
            byTitle: true,
            byDescription: false,
            search:''
        }
    }

    componentDidMount() {
        this.setState({loading: true})
        retrieveItem('token')
            .then(data => {
                this.setState({token: JSON.parse(data)})
                quizSearch(this.state.token, 0, 0, 0, this.state.byTitle ? 'title' : 'description', this.state.search)
                    .then((data) => {
                        console.log(data);
                        this.setState({data: data});
                        this.setState({loading: false})
                    })
                    .catch((error) => {
                        this.setState({error: error.message})
                        this.setState({loading: false})
                        removeAll()
                            .then((r) => {
                                if (r) {
                                    this.props.navigation.navigate('Auth');
                                }
                            });

                    })
            })
    }

    search() {
        this.setState({loading: true})
        quizSearch(this.state.token, 0, 0, 0, this.state.byTitle ? 'title' : 'description', this.state.search)
            .then((data) => {
                console.log(data);
                this.setState({data: data});
                this.setState({loading: false})
            })
            .catch((error) => {
                this.setState({error: error.message})
                this.setState({loading: false})
                removeAll()
                    .then((r) => {
                        if (r) {
                            this.props.navigation.navigate('Auth');
                        }
                    });

            })
    }

    _start = () => {
        return Animated.parallel([
            Animated.timing(this.state.slideUpValue, {
                toValue: this.state.showFilter ? 0 : 1 ,
                duration: 120,
                useNativeDriver: true
            })
        ]).start();
    };

    render() {
        let { slideUpValue} = this.state;

        const { navigation } = this.props;

        let showLoading = (
            this.state.loading ?
                <View style={[styles.containerLoading]} className='loadingShow'>
                    <ActivityIndicator animating={this.state.loading} size="large"  color="grey"/>
                </View> :
                <View></View>
        );

        let buttons = (
            WEB ?
                <View style={styles.containerButton}>
                    <View style={styles.buttonContainer}>
                        <Button icon={
                            <Icon
                                name="arrow-left"
                                size={18}
                                color="white"
                            />
                        }
                                buttonStyle={styles.buttonPrevNext}
                                className='add-button'
                                onPress={() => {
                                    this.refs.carousel.snapToPrev();
                                }}/>
                    </View>
                    <View style={styles.buttonContainer}>
                        <Button icon={
                            <Icon
                                name="arrow-right"
                                size={18}
                                color="white"
                            />
                        }
                                buttonStyle={styles.buttonPrevNext}
                                className='remove-button'
                                onPress={() => {
                                    this.refs.carousel.snapToNext();
                                }}/>
                    </View>
                </View>:
                <View></View>
        );

        return (
            <View style={styles.container}>
                <View style={styles.containerTitle}>
                    <Text h2 style={{fontWeight: '600'}}  >Quizzes</Text>
                </View>



                {showLoading}
                <Carousel
                    ref={'carousel'}
                    data={this.state.data}
                    renderItem=
                        {({item, index}) => (
                            <SliderEntry  navigation={navigation} data={item} even={(index + 1) % 2 === 0}/>
                        )}
                    sliderWidth={sliderWidth}
                    itemWidth={itemWidth}
                    layout={'stack'}
                    scrollInterpolator={ WEB ? scrollInterpolators[`scrollInterpolator1`] : null  }
                    slideInterpolatedStyle={ WEB ? animatedStyles[`animatedStyles1`] : null  }
                />
                {buttons}
                    <Animated.View
                        style={{

                            transform: [
                                {
                                    translateX: slideUpValue.interpolate({
                                        inputRange: [0, 1],
                                        outputRange: [WEB ? -2000 : -600, 0]
                                    }),
                                }
                            ],
                            width: 250,
                            height: 330,
                            backgroundColor: 'black',
                            marginTop: WEB ? 160 : 250,
                            alignSelf: 'center',
                            position: 'absolute',
                            borderRadius: 14,
                            shadowColor: "#000",
                            shadowOpacity: 0.60,
                            shadowOffset: { width: 0, height: 10 },
                            shadowRadius: 10,
                            elevation: 10,
                        }}
                    >
                        <Text  style={{color:'white', fontSize:30, alignSelf:'center', marginTop:30}}  >Filter</Text>
                        <Text  style={{color:'white', fontSize:22, marginTop:10, marginLeft:30}}  >Order by</Text>
                        <TouchableOpacity
                            style={{
                                flexDirection: "row",
                                //backgroundColor: "#192338",
                                justifyContent: "flex-start",
                                alignItems: "flex-start",
                                zIndex: -1,
                                borderRadius: 14}}
                            onPress={() => {
                                this.setState({byTitle: !this.state.byTitle});
                                this.setState({byDescription: !this.state.byDescription});
                            }
                            }
                        >
                            <Text style={{color:'white', fontSize:17, marginTop:10, marginLeft:30}}>  Title    {this.state.byTitle ?
                                <Icon
                                    name='check'
                                    size={18}
                                    color='#33d9b2'
                                />:
                                <Text/>
                            } </Text>
                        </TouchableOpacity>

                        <TouchableOpacity
                            style={{
                                flexDirection: "row",
                                //backgroundColor: "#192338",
                                justifyContent: "flex-start",
                                alignItems: "flex-start",
                                zIndex: -1,
                                borderRadius: 14}}
                            onPress={() => {
                                this.setState({byTitle: !this.state.byTitle});
                                this.setState({byDescription: !this.state.byDescription});
                            }
                            }
                        >
                            <Text style={{color:'white', fontSize:17, marginTop:10, marginLeft:30}}>  Description    {this.state.byDescription ?
                                <Icon
                                    name='check'
                                    size={18}
                                    color='#33d9b2'
                                />:
                                <Text/>
                            } </Text>
                        </TouchableOpacity>



                        <Text  style={{color:'white', fontSize:22, marginTop:20, marginLeft:30}}  >Search</Text>

                        <TextInput
                            className='title2'
                            style={styles.inputAns}
                            placeholder='Word'
                            placeholderTextColor='darkgrey'
                            onChangeText={val =>  this.setState({search: val})}
                        />

                        <TouchableOpacity
                            style={styles.button}
                            className='search-button'
                            onPress={() => {
                                this.setState({showFilter: !this.state.showFilter});
                                this._start();
                                this.search();
                            }}>
                            <Text  style={{color:'black', fontSize:15, alignSelf:'center', marginTop:6, fontWeight:'bold'}}  >OK</Text>
                        </TouchableOpacity>

                    </Animated.View>


                <ActionButton
                    className='create-button'
                    hideShadow={true}
                    buttonColor="black"
                    size={65}
                    renderIcon={active => active ? (<Icon className='filter-active' name="filter" size={30} color="white"/> ) : (<Icon className='filter-inactive'  name="filter" size={30} color="white"/>)}
                    onPress={() => {
                        this.setState({showFilter: !this.state.showFilter})
                        this._start()
                        }}
                />

            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        //justifyContent: 'center',
        alignItems: 'center',
        padding: 20
    },
    containerTitle: {
        marginTop: 50,
        marginBottom: 10,
        padding: 20,
    },


    containerSettings: {
        padding: 20,
    },
    button: {
        marginTop:20,
        alignSelf: 'center',
        width: 70,
        height: 30,
        backgroundColor: 'white',
        borderRadius: 14
    },
    buttonText: {
        fontSize: 20,
        color: 'grey'
    },
    buttonSO: {
        position: 'absolute',
        bottom: 0,
        marginBottom: 50,
    },
    containerButton: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        marginBottom:18
    },
    buttonContainer: {
        alignItems: 'center',
        justifyContent: 'center',
    },
    buttonPrevNext: {
        width: 120,
        height: 35,
        marginTop: 10,
        margin: 10,
        backgroundColor: 'black',
        borderRadius: 14
    },
    containerButtons: {
        flexDirection: 'row'
    },
    inputAns: {
        marginLeft:40,
        width: 150,
        height: 30,
        backgroundColor: 'dimgrey',
        margin: 10,
        padding: 8,
        color: 'white',
        borderRadius: 14,
        fontSize: 15,
        fontWeight: '500',
    },
    containerLoading: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        padding: 10
    },


})
