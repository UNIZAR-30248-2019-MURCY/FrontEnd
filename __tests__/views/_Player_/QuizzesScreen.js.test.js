import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import QuizzesScreen from "../../../app/views/Player/QuizzesScreen";
import {retrieveItem} from "../../../app/modules/AsyncStorage/retrieve";


describe('<QuizzesScreen/>', () => {
    it('QuizzesScreen renders without crashing', async () => {
        await retrieveItem();
        const rendered = renderer.create(<QuizzesScreen/>).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('CreateQuiz test against snapshot', async () => {
        await retrieveItem();
        const quizzesScreen = renderer.create(<QuizzesScreen />).toJSON();
        expect(quizzesScreen).toMatchSnapshot();
    });
    it('should render the Loading component if state.loading is true', async () => {
        await retrieveItem();

        let createComponent = shallow(<QuizzesScreen />);
        createComponent.setState({ loading: true });
        expect(createComponent.find('.loadingShow').length).toBe(1);
    });
    it('should not render the Loading component if state.loading is false', async () => {
        await retrieveItem();

        let createComponent = shallow(<QuizzesScreen />);
        createComponent.setState({ loading: false });
        expect(createComponent.find('.loadingShow').length).toBe(0);
    });
});

describe('.renderItem', () => {
    const mockItem = {
        title: 'Title',
    }
    let renderItemShallowWrapper;
    let wrapper = shallow(<QuizzesScreen/>);

    beforeAll(() => {
        let RenderItem = wrapper.find('.carousel').prop('renderItem');
        renderItemShallowWrapper = shallow(<RenderItem item={mockItem} />);
    });

    it('should match the snapshot', () => {
        expect(renderItemShallowWrapper).toMatchSnapshot();
    });
});
