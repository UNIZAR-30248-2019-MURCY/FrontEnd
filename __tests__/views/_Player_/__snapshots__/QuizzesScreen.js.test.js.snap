// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.renderItem should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`<QuizzesScreen/> CreateQuiz test against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 50,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 42.5,
          "fontWeight": "600",
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Quizzes
    </Text>
  </View>
  <View
    className="loadingShow"
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "padding": 10,
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="grey"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
  <RCTScrollView
    activeAnimationOptions={null}
    activeAnimationType="timing"
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDelay={1000}
    autoplayInterval={3000}
    callbackOffsetMargin={5}
    className="carousel"
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    contentContainerStyle={
      Array [
        Object {
          "paddingLeft": 67.5,
          "paddingRight": 67.5,
        },
        Object {},
      ]
    }
    data={Array []}
    decelerationRate="fast"
    directionalLockEnabled={true}
    enableMomentum={false}
    enableSnap={true}
    firstItem={0}
    hasParallaxImages={false}
    horizontal={true}
    inactiveSlideOpacity={0.7}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    inverted={false}
    itemWidth={615}
    layout="stack"
    lockScrollTimeoutDuration={1000}
    lockScrollWhileSnapping={false}
    loop={false}
    loopClonesPerSide={3}
    onLayout={[Function]}
    onResponderRelease={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    overScrollMode="never"
    pinchGestureEnabled={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollInterpolator={null}
    scrollsToTop={false}
    shouldOptimizeUpdates={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    slideInterpolatedStyle={null}
    slideStyle={Object {}}
    sliderWidth={750}
    style={
      Object {
        "flexDirection": "row",
        "opacity": 0,
        "width": 750,
      }
    }
    swipeThreshold={20}
    useScrollView={false}
    vertical={false}
  >
    <View />
  </RCTScrollView>
  <View />
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "black",
        "borderRadius": 14,
        "elevation": 10,
        "height": 330,
        "marginTop": 250,
        "position": "absolute",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.6,
        "shadowRadius": 10,
        "transform": Array [
          Object {
            "translateX": -600,
          },
        ],
        "width": 250,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontSize": 30,
          "marginTop": 30,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Filter
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "marginLeft": 30,
          "marginTop": 10,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Order by
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "borderRadius": 14,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "opacity": 1,
          "zIndex": -1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 17,
            "marginLeft": 30,
            "marginTop": 10,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
          Title    
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#33d9b2",
                "fontSize": 18,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ï€Œ
        </Text>
         
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "borderRadius": 14,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "opacity": 1,
          "zIndex": -1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 17,
            "marginLeft": 30,
            "marginTop": 10,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
          Description    
        <Text
          style={Object {}}
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          
        </Text>
         
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "marginLeft": 30,
          "marginTop": 20,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Search
    </Text>
    <TextInput
      allowFontScaling={true}
      className="title2"
      onChangeText={[Function]}
      placeholder="Word"
      placeholderTextColor="darkgrey"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "dimgrey",
          "borderRadius": 14,
          "color": "white",
          "fontSize": 15,
          "fontWeight": "500",
          "height": 30,
          "margin": 10,
          "marginLeft": 40,
          "padding": 8,
          "width": 150,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 14,
          "height": 30,
          "marginTop": 20,
          "opacity": 1,
          "width": 70,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "fontSize": 15,
            "fontWeight": "bold",
            "marginTop": 6,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        OK
      </Text>
    </View>
  </View>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "elevation": undefined,
            "justifyContent": "flex-end",
            "zIndex": undefined,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "elevation": undefined,
          "justifyContent": "flex-end",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": undefined,
        }
      }
    />
    <View
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": undefined,
              "justifyContent": "flex-end",
              "zIndex": undefined,
            },
          ],
          Object {
            "alignItems": "flex-end",
          },
          Object {
            "paddingVertical": 30,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginHorizontal": 30,
              "zIndex": undefined,
            },
            false,
            false,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 1)",
                "borderRadius": 32.5,
                "height": 65,
                "width": 65,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 32.5,
                  "height": 65,
                  "justifyContent": "center",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                  "width": 65,
                }
              }
            >
              <Text
                allowFontScaling={false}
                className="filter-inactive"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ï‚°
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
